From 4001b3de836a32ef4cbd880135dea6af6d8c3c52 Mon Sep 17 00:00:00 2001
From: Lukas Czerner <lczerner@redhat.com>
Date: Wed, 29 Oct 2014 16:34:08 -0400
Subject: [fs] ext4: remove readpage() check in ext4_mmap_file()

Message-id: <1414600485-1929-19-git-send-email-lczerner@redhat.com>
Patchwork-id: 98565
O-Subject: [RHEL 7.1 PATCH v2 18/55] ext4: remove readpage() check in ext4_mmap_file()
Bugzilla: 1156096
RH-Acked-by: Eric Sandeen <sandeen@redhat.com>

From: Zhang Zhen <zhenzhang.zhang@huawei.com>

BZ 1156096
https://bugzilla.redhat.com/show_bug.cgi?id=1156096

BREW 8169722
https://brewweb.devel.redhat.com/taskinfo?taskID=8169722

Upstream commit 590a141863f726216d452f26bdcc59d648fd7cb0

 There is no kind of file which does not supply a page reading function.

 Signed-off-by: Zhang Zhen <zhenzhang.zhang@huawei.com>
 Signed-off-by: Theodore Ts'o <tytso@mit.edu>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/ext4/file.c b/fs/ext4/file.c
index b46dfd4..a563e5d 100644
--- a/fs/ext4/file.c
+++ b/fs/ext4/file.c
@@ -206,10 +206,6 @@ static const struct vm_operations_struct ext4_file_vm_ops = {
 
 static int ext4_file_mmap(struct file *file, struct vm_area_struct *vma)
 {
- struct address_space *mapping = file->f_mapping;
-
- if (!mapping->a_ops->readpage)
-  return -ENOEXEC;
  file_accessed(file);
  vma->vm_ops = &ext4_file_vm_ops;
  return 0;
-- 
1.7.1